apps/keyboard/js/keyboard/layout_normalizer.js

```diff
@@ -23,7 +23,6 @@ var LayoutKeyNormalizer = function(page) {
 
 LayoutKeyNormalizer.prototype._isSpecialKey = function(key) {
   var SPECIAL_CODES = [
-    KeyEvent.DOM_VK_BACK_SPACE,
```

Comment:
`key.isSpecialKey` (once normalized) carries out whether the key is a special key.

[js/views/key_view.js](https://github.com/ssaavedra/gaia/blob/bugfix-1212588-keyboard-delete-role-key/apps/keyboard/js/views/key_view.js) controls how each key is painted. In concrete, isSpecialKey pushes the 'special-key' class and inhibits the role=key tag attribute from being set, no matter whether 'special-key' is set amongst the key's classes (which I suppose, semantically, it shouldn't).

If you still want backspace to be a special key then the logic should not be handled here, but instead by key_view.js.